### PR TITLE
[ENH] `collapse_levels`

### LINKED ISSUE #211: [ENH] make collapse_levels applicable to multilevel index
collapse levels of dataframes with multilevel index

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- add `glue` and `axis` parameters

**This PR resolves #211 .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1323--pyjanitor.netlify.app

File: tests/functions/test_collapse_levels.py [hunk 1]
```diff
@@ -63,3 +67,87 @@ def test_collapse_levels_functionality_3level(
             "decorated-elephantAsDfr.i.p-rhino :'(AsDfdeadly__flamingo",
         ]
     )
+
+
+def mklbl(prefix, n):
```
Comment (ericmjl) [inline tests/functions/test_collapse_levels.py]: Could you help me understand what `mklbl` stands for? Perhaps adding the intent of the function into the docstring?

Comment (samukweku) [reply tests/functions/test_collapse_levels.py]: nice catch! helps in creating a mutiindex dataframe. removing it, as it is a duplicate

**Comment by ericmjl** [comment]
Merged!

**Comment by ericmjl** [comment]
@samukweku would you like to do the honors of cutting a new release? I think the time is ripe!

**Comment by samukweku** [comment]
@ericmjl sure. if it is ok with you, i'll push a PR to resolve deprecations from pandas 2. then we can release a new version of pyjanitor?

**Pull request merged**